jenkins-setup/setup.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -e
+
+JENKINS_ROOT=./jenkins
+JENKINS_VERSION=2.319.3
+JENKINS_PM_VERSION=2.5.0
+JFR_VERSION=1.0-beta-30
+
+if [ $# == 1 ]
+then
+    JENKINS_VERSION=$1
+fi
+
+JENKINS_PM_URL=https://github.com/jenkinsci/plugin-installation-manager-tool/releases/download/${JENKINS_PM_VERSION}/jenkins-plugin-manager-${JENKINS_PM_VERSION}.jar
+JENKINS_CORE_URL=http://updates.jenkins.io/download/war/${JENKINS_VERSION}/jenkins.war
+JENKINS_JFR_URL=https://github.com/jenkinsci/jenkinsfile-runner/releases/download/${JFR_VERSION}/jenkinsfile-runner-${JFR_VERSION}.zip
```

Comment:
I think these variables should be overridable in case someone wants to use a different update center. Something like `FOO="${VARIABLE:=default}"` should work.

Reply:
> I think these variables should be overridable in case someone wants to use a different update center. Something like `FOO="${VARIABLE:=default}"` should work.

@AbhyudayaSharma Sorry, I don't fully understand your point. If you want users override these vars, it can only implemented by new parameters now. They cannot add changes directly at this shell script. Furthermore, can we find other update centers for these binaries? I only know plugins have different update centers.

Reply:
> @AbhyudayaSharma Sorry, I don't fully understand your point. If you want users override these vars, it can only implemented by new parameters now. 

@Cr1t-GYM The user should be able to set an environment variable that overrides these values. If an environment variable is set, use that. Otherwise use the default values. This is what I have in mind:

```yaml
- name: install plugins
  uses: plugin-installation-action
  env:
    JENKINS_CORE_URL: https://foo.example.com
```

Reply:
@AbhyudayaSharma I see. So what you want to do is the following codes. Is it correct?
```shell
JENKINS_PM_URL="${JENKINS_PM_URL:-https://github.com/jenkinsci/plugin-installation-manager-tool/releases/download/${JENKINS_PM_VERSION}/jenkins-plugin-manager-${JENKINS_PM_VERSION}.jar}"
JENKINS_CORE_URL="${JENKINS_CORE_URL:-http://updates.jenkins.io/download/war/${JENKINS_VERSION}/jenkins.war}"
JENKINS_JFR_URL="{$JENKINS_JFR_URL:-https://github.com/jenkinsci/jenkinsfile-runner/releases/download/${JFR_VERSION}/jenkinsfile-runner-${JFR_VERSION}.zip}"
```

Reply:
Yes, something like that. If this looks ugly, you can use `if [[ -n "$VAR" ]]`.